File: 07_fibonacci/fibonacci.js
function fibonacci(num) {
	if (num < 0) {
		return "Error: Number must be a positive integer.";
	} else if (num == 0) {
		return 0;
	}

	var previous = 0;
	var current = 1;

	for (var i = 1; i < num; i++){
		var temp = current;
		current += previous;
		previous = temp;
	}
	return current;
}

console.log(fibonacci(1));
console.log(fibonacci(2));
console.log(fibonacci(3));
console.log(fibonacci(4));
console.log(fibonacci(5));
console.log(fibonacci(6));
console.log(fibonacci(7));
console.log(fibonacci(8));

File: 07_fibonacci/fibonacci.py
def fibonacci(num):
	if num < 0:
		return "Error: Number must be a positive interger."
	elif num == 0:
		return 0

	previous = 0;
	current = 1;

	for idx in range(1, num):
		temp = current
		current += previous
		previous = temp

	return current


print(fibonacci(0))
print(fibonacci(1))
print(fibonacci(2))
print(fibonacci(3))
print(fibonacci(4))
print(fibonacci(5))
print(fibonacci(6))
print(fibonacci(7))
